src/codeu/chat/server/Controller.java

```diff
@@ -103,6 +100,107 @@ public Message newMessage(Uuid id, Uuid author, Uuid conversation, String body,
   }
 
   @Override
+  public boolean deleteMessage(Uuid msg, Uuid conversation) {
```

Comment:
Just for my clarity, this method finds the message then deletes it and the iterators put the conversation back in the same exact order except without the deleted message correct?